kernel/net/net_udp.c

```diff
@@ -633,15 +592,8 @@ static void net_udp_close(net_socket_t *hnd) {
     struct udp_pkt *pkt;
     struct udp_pkt *it;
 
-    if(irq_inside_int()) {
-        if(mutex_trylock(&udp_mutex) == -1) {
-            errno = EWOULDBLOCK;
-            return;
-        }
-    }
-    else {
-        mutex_lock(&udp_mutex);
-    }
+    if(mutex_lock_irqsafe(&udp_mutex))
```

Comment:
Seems odd that we're not indicating any error here (no -1 return or errno set)

Reply:
`errno` is set within `mutex_lock_irqsafe`.